### PR TITLE
Improve the error detection logic in Salt Provider

### LINKED ISSUE #2841: SaltProvider does not allow you to apply classes to HTML element when used in a new window
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

1.9.0

### Description

There are cases where a developer wants to apply the theme classes to the html element of a newly created window. However this is impossible at the moment because of the checks the salt provider peforms to see if it is the top level provider.

### Steps to reproduce

I used a window provider to pass the new window to the salt provider via context and passed in "root" to the applyClassesTo prop in the salt provider. However I get the following error message "SaltProvider can only apply CSS classes to the root if it is the root level SaltProvider."

### Expected behavior

I expected the classes to be applied to the html element of the newly created window.

### Operating system

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by joshwooding**
Closes #2841 

Updates the error detection logic so you can have a root provider in a new window.

### CODE REVIEW

File: packages/core/src/salt-provider/SaltProvider.tsx [hunk 1]
```diff
@@ -153,20 +164,13 @@ function InternalSaltProvider({
...
-  });
-
   useIsomorphicLayoutEffect(() => {
     const themeNames =
       themeName === DEFAULT_THEME_NAME
         ? [DEFAULT_THEME_NAME]
         : [DEFAULT_THEME_NAME, themeName];
     if (applyClassesTo === "root" && targetWindow) {
-      if (isRoot) {
+      if (isRoot && inheritedWindow != targetWindow) {
```
Comment (ImmanuelBaskaran) [inline packages/core/src/salt-provider/SaltProvider.tsx:173]: if isRoot is false, then this whole condition will go the the else branch?

**Comment by ImmanuelBaskaran** [comment]
Just confirmed the fix works.

**Review by ImmanuelBaskaran — APPROVED** [review]

**Review by origami-z — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Review by ImmanuelBaskaran — APPROVED** [review]

**Pull request merged**